netbout-web/src/main/java/com/netbout/rest/TkApp.java

```diff
@@ -90,16 +92,28 @@
      * @throws IOException If fails
      */
     public TkApp(final Base base) throws IOException {
-        super(TkApp.make(base));
+        this(base, new PsEmpty());
     }
 
     /**
      * Ctor.
      * @param base Base
+     * @param pass Last Pass to be executed
+     * @throws IOException If fails
+     */
+    public TkApp(final Base base, final Pass pass) throws IOException {
+        super(TkApp.make(base, pass));
+    }
+
+    /**
+     * Ctor.
+     * @param base Base
+     * @param pass Last Pass to be executed by the TkAppAuth
      * @return Take
      * @throws IOException If fails
      */
-    private static Take make(final Base base) throws IOException {
+    private static Take make(final Base base, final Pass pass)
```

Comment:
@antonini instead of passing `Pass`, we could pass `Optional<Pass>` - that way we get rid of the `instanceof` below which is not good.